CNAME

```diff
@@ -1 +1 @@
-envoyproxy.io
\ No newline at end of file
+www.envoyproxy.io
```

Comment:
@mattklein123 im not sure if this change is correct, or even if this file is required

`www.envoyproxy.io` is a `CNAME` (whereas `envoyproxy.io` is not)

```console
$ dig www.envoyproxy.io
...snip...
;; ANSWER SECTION:
www.envoyproxy.io.	300	IN	CNAME	teacher-glances-63085.netlify.com.
teacher-glances-63085.netlify.com. 20 IN A	3.67.255.218
teacher-glances-63085.netlify.com. 20 IN A	3.125.252.47
```

checking the netlify config and it is setup as expected (independently of any `CNAME`)

checking the history of this file and it has been repeatedly added/deleted but never any explanation as to why afaict

https://github.com/envoyproxy/envoyproxy.github.io/commits/main/CNAME

im pretty sure the other change in this PR is correct - which also leads me to think that if we need this file then it most likely should have been updated

Reply:
investigating this a bit further - this seems to be a jekyll thing for redirects or similar

im reasonably confident we can remove this file - ill do it in a separate PR so it can be easily reverted